tests/suites/test_suite_psa_crypto_metadata.function

```diff
@@ -56,6 +56,18 @@
 #define TEST_CLASSIFICATION_MACRO( flag, alg, flags )           \
     TEST_ASSERT( PSA_##flag( alg ) == !! ( ( flags ) & flag ) )
 
+/* Check the parity of value.
+ * Return 0 if value has even parity and a nonzero value otherwise. */
+int test_parity( uint32_t value )
+{
+    value ^= value >> 16;
+    value ^= value >> 8;
+    value ^= value >> 4;
+    return( 0x9669 & 1 << ( value & 0xf ) );
+}
+#define TEST_PARITY( value )                    \
```

Comment:
Suggestion: `TEST_ODD_PARITY()` to make it clear what about parity you are checking for.

Reply:
It's actually even parity (so that 0 has the correct parity). And I think the macro name should not indicate which parity it is. The job of the macro is to test for the _expected_ parity, not for even parity.

Reply:
I'll explain where I'm coming from. If a hardware datasheet says "odd parity bit", I would expect a value of 1 there to mean odd parity and 0 to mean even parity. If a datasheet says "even parity bit", then I expect a value of 1 there to mean even parity, and 0 to mean odd parity. If the datasheet says "parity bit", I am unclear what the hardware will place there.

In this case, I read "value" as the value you wanted to check the parity of, not the expected parity. Consider renaming "value" to "expected_parity" if the job of the macro is to test for arbitrary expected parity. Consider renaming to `TEST_EVEN_PARITY` if value is the value you are checking the parity of. The comment you've added to `test_parity` is good, as it says what sort of parity you are checking for, but the names could still be more clear.

Reply:
The name of the macro deliberately doesn't say which way the parity should go because its job is to test for whatever parity the API design intends. The fact that it's even parity is an implementation detail.

I don't understand how you come to name it odd parity. A value is correct iff the number of bits that is set is even.

I don't understand why you want to rename the argument. The argument is a value whose parity needs to be checked. There is no argument indicating what parity is expected: that's built into the code, since all values that are checked for parity must obey the same parity requirement.